src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -209,28 +199,34 @@ internal static bool IsRuntimeAssembly(string assemblyPath)
 
         internal static bool IsRuntimeAssembly(AssemblyName assemblyName)
         {
-            var publicKeyToken = assemblyName.GetPublicKeyToken();
-            var lowerInvariant = BitConverter.ToString(publicKeyToken).Replace("-", string.Empty).ToLowerInvariant();
+            var keyToken = assemblyName.GetPublicKeyToken();
+            if (keyToken.Length == 0)
+            {
+                return false;
+            }
+
+            var publicKeyToken = BitConverter.ToInt64(keyToken, 0);
+            
             //System
-            if (lowerInvariant == "b77a5c561934e089")
+            if (publicKeyToken == -8511746012302509385)
             {
                 return true;
             }
 
             //System.Core +  mscorelib
-            if (lowerInvariant == "7cec85d7bea7798e")
```

Comment:
how do I get this one? mscorlib and System.Core seem to have the same public key as system.dll (b77a5c561934e089)